fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java

```diff
@@ -373,10 +373,8 @@ public synchronized void replayEraseTable(long tableId) {
                 Runnable runnable = null;
                 Table table = tableInfo.getTable();
                 nameToTableInfo.remove(dbId, table.getName());
-                if (!isCheckpointThread()) {
-                    runnable = table.delete(true);
-                }
-                if (runnable != null) {
+                runnable = table.delete(true);
```

Comment:
I'm not familiar with the code logic here, so it's safe to call `table.delete()` from  checkpoint thread now?

Reply:
I think so. As I wrote in the description of this PR, it eventually call `TabletInvertedIndex#deleteTablet` function which will do nothing if current thread is checkpoint thread. So there is no need to check it here.

Reply:
Got it, thx.